### PR TITLE
add the pg version 16

### PR DESCRIPTION
**Pull request by jopadi**
add the changes go add the new pg version 16 and remove the pg10

- [x]  include 16 in CRDs
- [x]  remove 10 
- [x]  bump target_major_version from "15" to "16"
- [x]  update some unit test examples to use version 16
- [x]  update postgres doc links to 16
- [x]  update the spilo image to pg16

### CODE REVIEW

**Comment by FxKu** [comment]
Lets bump `minimal_major_version`, too. Maybe 12 or even 13.

File: e2e/run.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ IFS=$'\n\t'
 
 readonly cluster_name="postgres-operator-e2e-tests"
 readonly kubeconfig_path="/tmp/kind-config-${cluster_name}"
-readonly spilo_image="registry.opensource.zalan.do/acid/spilo-15-e2e:0.1"
+readonly spilo_image="registry-write.opensource.zalan.do/acid/spilo-16-e2e:0.1"
```
Comment (FxKu) [inline e2e/run.sh:11]:
```suggestion
readonly spilo_image="registry.opensource.zalan.do/acid/spilo-16-e2e:0.1"
```

File: charts/postgres-operator/crds/postgresqls.yaml [hunk 2]
```diff
@@ -371,12 +371,11 @@ spec:
                   version:
                     type: string
                     enum:
-                      - "10"
-                      - "11"
```
Comment (FxKu) [inline charts/postgres-operator/crds/postgresqls.yaml:375]: we still need 11 here

File: manifests/postgresql.crd.yaml [hunk 3]
```diff
@@ -369,8 +369,6 @@ spec:
                   version:
                     type: string
                     enum:
-                      - "10"
-                      - "11"
```
Comment (FxKu) [inline manifests/postgresql.crd.yaml:373]: we still need 11 here

File: pkg/apis/acid.zalan.do/v1/crds.go [hunk 4]
```diff
@@ -589,12 +589,6 @@ var PostgresCRDResourceValidation = apiextv1.CustomResourceValidation{
 							"version": {
 								Type: "string",
 								Enum: []apiextv1.JSON{
-									{
-										Raw: []byte(`"10"`),
-									},
-									{
-										Raw: []byte(`"11"`),
-									},
```
Comment (FxKu) [inline pkg/apis/acid.zalan.do/v1/crds.go:597]: we still need 11 here

File: pkg/cluster/majorversionupgrade.go [hunk 5]
```diff
@@ -11,8 +11,6 @@ import (
 
 // VersionMap Map of version numbers
 var VersionMap = map[string]int{
-	"10": 100000,
-	"11": 110000,
```
Comment (FxKu) [inline pkg/cluster/majorversionupgrade.go:15]: we still need 11 here

File: README.md [hunk 6]
```diff
@@ -29,13 +29,13 @@ pipelines with no access to Kubernetes API directly, promoting infrastructure as
 
 ### PostgreSQL features
 
-* Supports PostgreSQL 15, starting from 10+
+* Supports PostgreSQL 16, starting from 12+
```
Comment (FxKu) [inline README.md:32]:
```suggestion
* Supports PostgreSQL 16, starting from 11+
```

**Comment by jopadi** [comment]
👍

**Comment by FxKu** [comment]
:+1:

**Comment by jopadi** [comment]
👍

**Pull request merged**